test/functional/users.js

```diff
@@ -902,6 +902,104 @@ describe("UsersController", function() {
     })
   })
 
+  describe('#updateProfileFrontendPreferences', function() {
+    var authToken
+      , user
+
+    beforeEach(funcTestHelper.createUser('Luna', 'password', function (token, luna) {
+      authToken = token
+      user = luna
+    }))
+
+    it('should store user preferences in db', async () => {
+      let prefs = {
+        'net.freefeed': {
+          'userProperty': 'value'
+        },
+        'custom.domain': {
+          'customProperty': 'value2'
+        }
+      }
+      {
+        let response = await funcTestHelper.updateUserAsync({ user, authToken },
+          { frontendPreferences: prefs }
+        )
+        response.status.should.eql(200)
+
+        let responseData = await response.json()
+        responseData.should.have.property('users')
+        responseData.users.should.have.property('frontendPreferences')
+        responseData.users.frontendPreferences.should.have.property('net.freefeed')
+        responseData.users.frontendPreferences['net.freefeed'].should.have.property('userProperty')
```

Comment:
4 lines can be replaced with just one:
`responseData.should.have.deep.property('users.frontendPreferences.net\\.freefeed.userProperty')`